apis/logging/v2beta/doc.go

```diff
@@ -0,0 +1,53 @@
+// package v2beta custom resorces for the cluster logging operator.
+//
+// NOTE: This API is under construction and subject to change.
+//
+// NOTE: Doc comments are written using JSON names (lowerCase) rather than Go UpperCase stlye.
+//
+// IMPORTANT: Run "make generate" to regenerate code after modifying any files in this package.
+// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+//
+// +groupName=logging.openshift.io
+// +k8s:deepcopy-gen=package,register
+// +k8s:openapi-gen=true
+// +kubebuilder:object:generate=true
+package v2beta
+
+/* FIXME: NOTES and OPEN QUESTIONS
+
+   # Namespace
+
+   A cluster-scoped ClsterLogForwarder needs a namespace for:
+   - serviceaccount
+   - configmap
+   - secrets
+   - deployments, daemonset, pods etc.
+
+   The namespace field is pulled to the top of the CLF spec since it applies to more than the SA.
+
+   Namespace ownership/lifecycle? Can the CLF own the namepsace?
+   If the SA is provided by the user, then we don't want to delete the namespace (and therefore the SA) when the forwarder is deleted.
+
+   Given the tight association between CLF and this namespace, should we reconsider making it cluster-scoped?
```

Comment:
Alternatively we could drop the notion of "ClusterLogForwarder" for "LogForwarder" but it seems awkward to continue to allow the namespace object access to cluster wide logs.  Is there any precedence for the construct we are proposing?

Reply:
Need to find out - there must be cluster-wide resources that create Pods etc. so there must be one somewhere.